pkg/maps/ctmap/ctmap.go

```diff
@@ -510,6 +517,108 @@ func GC(m *Map, filter *GCFilter) int {
 	return doGC(m, filter)
 }
 
+type CTMapIPVersion int
+
+const (
+	CTMapIPv4 CTMapIPVersion = iota
+	CTMapIPv6
+)
+
+// FilterMapsByProto filters the given CT maps by the given IP version, and
+// returns two maps - one for TCP and one for any protocol.
+func FilterMapsByProto(maps []*Map, ipVsn CTMapIPVersion) (ctMapTCP *Map, ctMapAny *Map) {
+	for _, m := range maps {
+		switch ipVsn {
+		case CTMapIPv4:
+			switch m.mapType {
+			case mapTypeIPv4TCPLocal, mapTypeIPv4TCPGlobal:
+				ctMapTCP = m
+			case mapTypeIPv4AnyLocal, mapTypeIPv4AnyGlobal:
+				ctMapAny = m
+			}
+		case CTMapIPv6:
+			switch m.mapType {
+			case mapTypeIPv6TCPLocal, mapTypeIPv6TCPGlobal:
+				ctMapTCP = m
+			case mapTypeIPv6AnyLocal, mapTypeIPv6AnyGlobal:
+				ctMapAny = m
```

Comment:
Logic looks right. We might be able to fold it in with the maptype helpers in `pkg/maps/ctmap/types.go` for better code reuse but it's not such a big deal.